gitrs: init at 0.3.6

mccurdyc opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

- Adds gitrs at v0.3.6 
- https://github.com/mccurdyc/gitrs
- "A simple, opinionated, tool, written in Rust, for declaratively managing Git repos on your machine."

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Twey committed:
makeWrapper: fix flag handling

When `--add-flags` is not used, `flagsBefore` is unset.  This causes an error when invoking `makeWrapper` from a context that sets `-o nounset`, as is done in `buildDotnetModule`.
This change makes `makeWrapper` safe for use in these conditions.

Szczyp committed:
brewtarget: 2.3.1 -> 3.0.5

lopsided98 committed:
ffmpeg: remove references to build platform compiler

When cross-compiling, the build platform compiler ends up in the runtime
closure because it is included in the configure flags that get embedded in the
binaries. To fix this, add the compiler to the remove-references-to invocation.

gaykitty committed:
kitty-img: init at 1.0.0

gaykitty committed:
fre: init at 0.3.1

oluceps committed:
graphite-cursors: init at 2021-11-26

poelzi committed:
kicli: init at 0.0.3

Commit:
tlaplus: remove jdk dependency

As of b5d3526, this package fetches a pre-compiled TLA+ distribution
instead of compiling from source.

Commit:
add myself to the maintainers list

This is my second patch to the tlaplus package and I agreed to add
myself as a maintainer on my next contribution.

corngood committed:
nixos/gnupg: add systemd configuration

This depended on the systemd user configuration provided upstream in
doc/examples.  However, this was all removed in:

https://github.com/gpg/gnupg/commit/eae28f1bd4a5632e8f8e85b7248d1c4d4a10a5ed

corngood committed:
gnupg: fix tests

quentinmit committed:
python311Packages.routeros-api: init at 0.17.0

poelzi committed:
arduinoOTA: init at 1.4.1

markuskowa committed:
hdf5: 1.14.0 -> 1.14.1-2

carlthome committed:
maintainers: add carlthome

carlthome committed:
python3Packages.mir_eval: init at 0.7

mrkkrp committed:
nushellPlugins.gstat: init

emilytrau committed:
minimal-bootstrap: create top-level attr for bootstrap sources

willcohen committed:
binaryen: 112 -> 113

reckenrode committed:
rav1e: fix build with updated Darwin stdenv

The updated Darwin stdenv uses `llvm-strip`, but that causes issues for
rav1e when using NASM and linking it with the cctools ld64. Stripping
the debug information instead of everything fixes the problem.

lheckemann committed:
glibc: split getent into its own output

Many dependents only require getent and not all the locale generation
tools that are included in the `bin` output. This can save some
closure size!

reckenrode committed:
apple_sdk_11_0.stdenv: replace extraBuildInputs with CF from 11.0 SDK

This can result in swift-corefoundation being included even when a
derivation uses `apple_sdk_11_0.stdenv` directly, which may result in
hard to debug compiler errors (unless you just happen to know cryptic
compiler errors means the 10.12 and 11.0 SDKs have been mixed).

reckenrode committed:
qt5.qtbase: fix build with Darwin sandbox enabled

Qt requires access to the system ICU data due to its linking against
the system CoreFoundation and invoking CF APIs that tries to access it.
This manifests as a crash during build when it fails to access the data.

reckenrode committed:
qtpass: fix build with Darwin sandbox enabled

Co-authored-by: toonn <toonn@toonn.io>

reckenrode committed:
qtkeychain: fix build with Darwin sandbox enabled

Co-authored-by: toonn <toonn@toonn.io>

nicoonoclaste committed:
below: Ship systemd service unit

nicoonoclaste committed:
nixos/below: Add service module

nicoonoclaste committed:
nixos/below: Explicitely mark descriptions as using Markdown

emattiza committed:
update: otel-cli 0.2.0 -> 0.3.0

updates otel-cli from 0.2.0 to 0.3.0
see https://github.com/equinix-labs/otel-cli/releases/tag/v0.3.0
for changelog and release details

trofi committed:
llvm_{10..13}: backport gcc-13 fixes

Without the change `llvm` build fails on `gcc-13` fails as:

    [  0%] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/Signals.cpp.o
    In file included from llvm/lib/Support/Signals.cpp:14:
    llvm/include/llvm/Support/Signals.h:119:8: error: variable or field 'CleanupOnSignal' declared void
      119 |   void CleanupOnSignal(uintptr_t Context);
          |        ^~~~~~~~~~~~~~~

GenericNerdyUsername committed:
gst_all_1.gst-plugins-bad: add opencvSupport option

gst-plugins-bad is used by a lot of packages, and opencv has `enableCuda` option. Forcing people to only use opencv when necessary reduces rebuild size when `config.cudaSupport = true`

willcohen committed:
emscripten: 3.1.24 -> 3.1.41

r-burns committed:
libclc: fix .pc paths

r-ryantm committed:
imath: 3.1.7 -> 3.1.9

willcohen committed:
llvmPackages_16: 16.0.1 -> 16.0.6

alyssais committed:
rustPlatform.cargoBuildHook: don't let cargo strip

This fixes debug info of cloud-hypervisor, which recently added
strip = true to its release profile in Cargo.toml.

tjni committed:
cargo-c: 0.9.13 -> 0.9.20

onny committed:
cups: 2.4.3 -> 2.4.5

Diff: https://github.com/OpenPrinting/cups/compare/None...v2.4.4

Artturin committed:
gnu-efi: fix musl build

gnu-efi> /build/gnu-efi-3.0.17//lib/console.c:84:68: error: passing argument 2 of ‘ConOut->OutputString’ from incompatible pointer type [8;;h
ttps://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
gnu-efi>    84 |                 uefi_call_wrapper(ConOut->OutputString, 2, ConOut, L"\b \b");
gnu-efi>       |                                                                    ^~~~~~~~
gnu-efi>       |                                                                    |
gnu-efi>       |                                                                    short unsigned int *

ajs124 committed:
apparmor: 3.1.4 -> 3.1.5

https://gitlab.com/apparmor/apparmor/-/wikis/Release_Notes_3.1.5

tie committed:
libgdiplus: 6.0.5 -> 6.1

AndersonTorres committed:
gtk2: refactor

amarshall committed:
fetchfirefoxaddon: formatting

Had some inconsistent spacing, so ran against nixpkgs-fmt and also split
fn args into multiple lines for simpler diffs.

amarshall committed:
fetchfirefoxaddon: make reproducible

r-ryantm committed:
hwdata: 0.370 -> 0.371

amarshall committed:
bisq-desktop: formatting

amarshall committed:
bisq-desktop: make reproducible

amarshall committed:
pcsx2: make reproducible

amarshall committed:
slimmerjs: formatting

amarshall committed:
mari0: formatting

amarshall committed:
slimmerjs: make reproducible

amarshall committed:
mari0: make reproducible

amarshall committed:
orthorobot: formatting

amarshall committed:
orthorobot: make reproducible

amarshall committed:
wireworld: formatting

amarshall committed:
wireworld: make reproducible

amarshall committed:
pridefetch: formatting

amarshall committed:
pridefetch: make reproducible

amarshall committed:
pcsx2: formatting

AndersonTorres committed:
cmake: 3.25.3 -> 3.26.4

trofi committed:
xterm: 380 -> 382

Changes: https://invisible-island.net/xterm/xterm.log.html#xterm_382

delroth committed:
mupdf: actually build and install the shared libraries version

The current version of the derivation builds in shared libraries mode,
but then the "shared" flag is not passed to the "make install"
invocation. This causes "make install" to perform a whole second build
from scratch, in static mode, and install this to $out instead.

Instead pass shared=yes as part of the makeFlags -- this is basically
the only thing that the "shared" build target does anyway, and I don't
think there is a similar target for "make install".

This issue was detected because the .pc shipped with mupdf doesn't work
with the static libraries currently shipped (it doesn't include
recursive dependencies like zlib).

corngood committed:
gnupg: fix test attribute key

reckenrode committed:
darwin.libmalloc: add at 116.50.8

reckenrode committed:
darwin.libmalloc: add at 317.40.8 for the 11.0 SDK

reckenrode committed:
darwin.Libsystem: get malloc.h from darwin.libmalloc

marsam committed:
bundler: 2.4.13 -> 2.4.14

Changelog: https://github.com/rubygems/rubygems/blob/bundler-v2.4.14/bundler/CHANGELOG.md

marsam committed:
ruby: rubygems 3.4.13 -> 3.4.14

Changelog: https://github.com/rubygems/rubygems/releases/tag/v3.4.14

reckenrode committed:
darwin.xnu: provide additional headers needed by Libsystem

reckenrode committed:
darwin.Libsystem: get asl.h from upstream syslog sources

reckenrode committed:
darwin.Libsystem: get pthread headers from darwin.libpthread

Some of the headers are symlinked to the root `include` in upstream
Libsystem, so also symlink them there from `include/pthread`.

reckenrode committed:
darwin.Libc: stop vendoring headers from other packages

A number of headers in Libc are being vendored from other packages.
Instead of copying them from an earlier Libc, Libsystem now sources them
from their respective packages (see below). This allows Libc_old to be
dropped and avoids any potential clashes when building Libsystem.

libmalloc:
* malloc/malloc.h

libplatform:
* setjmp.h
* ucontext.h
* libkern/OSAtomic.h
* libkern/OSCacheControl.h

libpthread:
* pthread*.h
* sched.h
* spawn.h

syslog (vendored because only one file is needed):
* asl.h

xnu:
* spawn.h (a different one from libpthread)
* libproc.h

K900 committed:
python3Packages.pydantic: fix doctests race

NickCao committed:
libopus: enable intrinsics only on supported platforms

tjni committed:
python310Packages.hatch-fancy-pypi-readme: 22.8.0 -> 23.1.0

endgame committed:
dhcpcd: Fetch source from GitHub

endgame committed:
openresolv: Fetch source from GitHub

endgame committed:
mirrors: Remove `roy`

Roy Marples now releases his code via GitHub.

raboof committed:
Revert "gcc: install info files serially"

Because the build still uses a pre-4.4 version of GNU Make,
this option makes the whole build serial rather than only
the installation.

See also https://gcc.gnu.org/PR109898

This reverts commit f3995cee01fad5aeb5b67de8426a7a0321c74ebf.

SuperSandro2000 committed:
python310.pkgs.f5-icontrol-rest: init at 1.3.15

SuperSandro2000 committed:
python310.pkgs.f5-sdk: init at 3.0.21

SuperSandro2000 committed:
python310.pkgs.netapp-lib: init at 2021.6.25

SuperSandro2000 committed:
python310.pkgs.cliche: init at 0.10.108

SuperSandro2000 committed:
python310.pkgs.recline: init at 2023.5

SuperSandro2000 committed:
python310.pkgs.netapp-ontap: init at 9.12.1.0

raboof committed:
gcc: disable parallelism when installing

to avoid missing sections in the info pages due
the index being overwritten by parallel `install-info`
invocations

fabianhjr committed:
openjdk17: 17.0.6+10 → 17.0.7+7

fabianhjr committed:
openjdk11: 11.0.18+10 → 11.0.19+7

fabianhjr committed:
adoptopenjdk: 8.0.322+6 → 8.0.372+7, 11.0.16+101 → 11.0.19+7, 17.0.4+101 → 17.0.7+7

fabianhjr committed:
jetbrains.jdk: 17.0.6-b829.9 → 17.0.7-b829.16

06kellyjac committed:
semgrep{,-core}: 1.15.0 -> 1.27.0

Now fetching semgrep-core from the python wheel as r2c don't add
binaries to the GH releases anymore.

tjni committed:
python310Packages.respx: patch to support httpx 0.24

alyssais committed:
linux: set ZPOOL=y

ZBUD depends on ZPOOL.  If we get asked about ZPOOL first, by default
it would be set to m, and it would then not be possible to set ZBUD=y.

This fixes, for example, building Linux 5.15 for microblazeel-linux.

zzzsyyy committed:
libjxl: 0.8.1 -> 0.8.2

corngood committed:
nixos/gpg-agent: move pinentry-program to /etc/gnupg/gpg-agent.conf

NickCao committed:
liburing: 2.3 -> 2.4

NickCao committed:
liburing: properly set configureFlags

NickCao committed:
liburing: add nickcao to maintainers

Ma27 committed:
python3.pkgs.invoke: install shell completions

Commit:
stdenv: updateAutotoolsGnuConfigScriptsHook unconditionally

Unlike autoreconfHook, updateAutotoolsGnuConfigScriptsHook adds
almost no compilations.  Therefore, in the interest of building the
same source code on every platform wherever possible, let's
eliminate the conditional guards around
updateAutotoolsGnuConfigScriptsHook in stdenv.

Commit:
glibc: allow users of glibc/common.nix to override makeFlags

This commit allows to include `makeFlags` in a glibc derivation
without clobbering the flags from `common.nix`

Commit:
glibcLocales: use more than one core to build

This commit massively reduces the build latency for glibcLocales by
allowing it to build in parallel.  This requires passing
`-j$NIX_BUILD_CORES` via the glibc-specific make variable
`PARALLELMFLAGS`.

This commit also fixes a preexisting bug where the glibcLocales
package would ignore `preBuild` and `postBuild`.

Commit:
glibcInfo: use makeFlags instead of buildPhase

This commit causes glibc/info.nix to use the standard builder's
`buildPhase` so things like `preBuild`, `postBuild`,
`enableParallelBuilding`, etc work correctly.

marsam committed:
python310Packages.pydantic: 1.10.8 -> 1.10.9

Diff: https://github.com/pydantic/pydantic/compare/refs/tags/v1.10.8...v1.10.9

Changelog: https://github.com/pydantic/pydantic/blob/v1.10.9/HISTORY.md

marsam committed:
maturin: 1.0.1 -> 1.1.0

Diff: PyO3/maturin@v1.0.1...v1.1.0

Changelog: https://github.com/PyO3/maturin/blob/v1.1.0/Changelog.md

Commit:
stdenv: eliminate duplicate gnu-config in extraNativeBuildInputs

Accidentally omitted from #237968.

fabianhjr committed:
xorg.libX11: 1.8.4 → 1.8.6

Ma27 committed:
rpcsvc-proto: 1.4.3 -> 1.4.4

ChangeLog: https://github.com/thkukuk/rpcsvc-proto/releases/tag/v1.4.4

r-ryantm committed:
rust-bindgen-unwrapped: 0.65.1 -> 0.66.0

kirillrdy committed:
icu: 73.1 -> 73.2

Commit:
minimal-bootstrap: make sources a non-tarballs.nixos.org FOD

This commit adjusts #232576 to break the
fetchurl<->minimal-bootstrap-sources dependency cycle without
needing an upload to tarballs.nixos.org.  It does this by appending
the low-level FOD attributes onto the `runCommand` derivation.

  https://nixos.org/manual/nix/unstable/language/advanced-attributes.html#adv-attr-outputHash

OPNA2608 committed:
libopenmpt: 0.7.1 -> 0.7.2

RaitoBezarius committed:
cryptsetup: re-encrypt is a default extension now

It can only be disabled.

RaitoBezarius committed:
cryptsetup: compile against libargon2 by default

This will provide a nice fast Argon2 implementation rather than the slow internal one.

RaitoBezarius committed:
cryptsetup: add raitobezarius as a maintainer

Here we go again. :)

witchof0x20 committed:
xivlauncher: 1.0.3->1.0.4

primeos committed:
chromiumDev: 116.0.5817.0 -> 116.0.5829.0

primeos committed:
chromiumBeta: 115.0.5790.24 -> 115.0.5790.32

SuperSandro2000 committed:
python311Packages.sqlalchemy-migrate: fix build

trofi committed:
glibcLocales: enable parallel building

alyssais committed:
meson: 1.1.0 -> 1.1.1

emilytrau committed:
caeml: init at unstable-2023-05-24

erikarvstedt committed:
systemd: fix services not stopping

Fixes #237591

NickCao committed:
fastJson: cleanup, use autoreconfHook

mweinelt committed:
python310Packages.wxPython_4_2: 4.2.0 -> 4.2.1

https://github.com/wxWidgets/Phoenix/blob/wxPython-4.2.1/CHANGES.rst

NickCao committed:
wasm-pack: 0.11.1 -> 0.12.0

Diff: https://github.com/rustwasm/wasm-pack/compare/v0.11.1...v0.12.0

lilyinstarlight committed:
gst_all_1.gstreamer: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-plugins-base: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-plugins-good: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-plugins-bad: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-plugins-ugly: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-libav: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-vaapi: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-rtsp-server: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-devtools: 1.22.3 -> 1.22.4

lilyinstarlight committed:
gst_all_1.gst-editing-services: 1.22.3 -> 1.22.4

lilyinstarlight committed:
python3Packages.gst-python: 1.22.3 -> 1.22.4

amarshall committed:
bitwarden: 2023.5.0 -> 2023.5.1

mweinelt committed:
python310Packages.asyncssh: 2.13.1 -> 2.13.2

https://github.com/ronf/asyncssh/blob/v2.13.2/docs/changes.rst

K900 committed:
sdl2: 2.26.5 -> 2.28.0

ajs124 committed:
proxysql: 2.5.2 -> 2.5.3

https://github.com/sysown/proxysql/releases/tag/v2.5.3

vcunat committed:
e2fsprogs: avoid incompatible features in mke2fs

by a patch taken from Debian's 1.47.0-2.  Discussion:
https://github.com/NixOS/nixpkgs/pull/229454#issuecomment-1601698426

K900 committed:
glslang: 12.1.0 -> 12.2.0

Diff: https://github.com/KhronosGroup/glslang/compare/12.1.0...12.2.0

Also drop now-unnecessary workarounds.

K900 committed:
vulkan-headers: 1.3.249 -> 1.3.254

Diff: https://github.com/KhronosGroup/Vulkan-Headers/compare/v1.3.249...v1.3.254

K900 committed:
vulkan-loader: 1.3.249 -> 1.3.254

Diff: https://github.com/KhronosGroup/Vulkan-Loader/compare/v1.3.249...v1.3.254

K900 committed:
vulkan-validation-layers: 1.3.249 -> 1.3.254

Diff: https://github.com/KhronosGroup/Vulkan-ValidationLayers/compare/v1.3.249...v1.3.254

K900 committed:
vulkan-tools: 1.3.249 -> 1.3.254

Diff: https://github.com/KhronosGroup/Vulkan-Tools/compare/v1.3.249...v1.3.254

Also add new Python dependency.

K900 committed:
vulkan-tools-lunarg: 1.3.249 -> 1.3.250

Diff: https://github.com/LunarG/VulkanTools/compare/v1.3.249...v1.3.250

Apply small hacks to make it build with latest SDK/VVL.

K900 committed:
vulkan-extension-layer: 1.3.248 -> 1.3.254

Diff: https://github.com/KhronosGroup/Vulkan-ExtensionLayer/compare/v1.3.248...v1.3.254

K900 committed:
spirv-headers: 1.3.243.0 -> 1.3.250.0

Diff: https://github.com/KhronosGroup/SPIRV-Headers/compare/sdk-1.3.243.0...sdk-1.3.250.0

Also remove now-unnecessary workaround.

K900 committed:
spirv-cross: 1.3.243.0 -> 1.3.250.0

Diff: https://github.com/KhronosGroup/SPIRV-Cross/compare/sdk-1.3.243.0...sdk-1.3.250.0

Changelog: https://github.com/KhronosGroup/SPIRV-Cross/releases/tag/23.11pre-git

K900 committed:
spirv-tools: 2023.2 -> 2023.3.rc1

Matches 1.3.250 SDK tag.

chvp committed:
eid-mw: 5.1.10 -> 5.1.11

https://github.com/Fedict/eid-mw/releases/tag/v5.1.11 (no changelog/release
notes, sadly)

K900 committed:
mesa: 23.1.2 -> 23.1.3

Changelog: https://www.mesa3d.org/relnotes/23.1.3.html

figsoda committed:
vimPlugins.flash-nvim: init at 2023-06-23

figsoda committed:
vimPlugins: update

figsoda committed:
vimPlugins.nvim-treesitter: update grammars

Kranzes committed:
wl-screenrec: init at unstable-2023-05-31

otavio committed:
tmuxp: fix shell completion generation

tmuxp 1.17+ uses shtab to generate the shell completion. This commit
adapts the code to properly generate the shell completion module.

A side effect of use of shtab is that fish is not supported so it has
been removed from postInstall.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

aaronjheng committed:
notation: init at 1.0.0-rc.7

otavio committed:
tmuxp: remove pypkgs binding to avoid indirection

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

r-ryantm committed:
lego: 4.12.1 -> 4.12.3

rnhmjoj committed:
compressFirmwareXz: preserve meta attributes

Among other things, this preserves the package priority, which is
important when building the `hardware.firmware` environment in NixOS.

figsoda committed:
deltachat-desktop: remove noop override

happysalada committed:
atomic-swap: init at 0.4.0

qbit committed:
tidal-hifi: 5.2.0 -> 5.3.0

figsoda committed:
tela-icon-theme: 2023-02-03 -> 2023-06-25

Diff: https://github.com/vinceliuice/tela-icon-theme/compare/2023-02-03...2023-06-25

Changelog: https://github.com/vinceliuice/Tela-icon-theme/releases/tag/2023-06-25

figsoda committed:
karmor: fix version

figsoda committed:
karmor: add version test

Atemu committed:
linux_xanmod_latest: 6.3.5 -> 6.3.9

Atemu committed:
linux_xanmod: 6.1.31 -> 6.1.35

BenediktBroich committed:
blueprint-compiler: 0.6.0 -> 0.8.1

BenediktBroich committed:
textpieces: mark broken

huantianad committed:
unityhub: add 2019 editor dependencies

Fixes #236381

huantianad committed:
unityhub: 3.4.2 -> 3.5.0

Commit:
maintainers: add earthengine

nevivurn committed:
gnucash: 5.1 -> 5.2

marsam committed:
esbuild: 0.18.8 -> 0.18.9

Diff: https://github.com/evanw/esbuild/compare/v0.18.8...v0.18.9

Changelog: https://github.com/evanw/esbuild/blob/v0.18.9/CHANGELOG.md

Commit:
yaydl: init at 0.13.0

fabaff committed:
python311Packages.env-canada: 0.5.34 -> 0.5.35

Diff: https://github.com/michaeldavie/env_canada/compare/refs/tags/v0.5.34...v0.5.35

Changelog: https://github.com/michaeldavie/env_canada/blob/v0.5.35/CHANGELOG.md

fabaff committed:
python311Packages.fastapi-mail: 1.2.8 -> 1.3.0

Changelog: https://github.com/sabuhish/fastapi-mail/releases/tag/1.3.0

fabaff committed:
python311Packages.hahomematic: 2023.6.0 -> 2023.6.1

Diff: https://github.com/danielperna84/hahomematic/compare/refs/tags/2023.6.0...2023.6.1

Changelog: https://github.com/danielperna84/hahomematic/releases/tag/2023.6.1

r-ryantm committed:
libgbinder: 1.1.33 -> 1.1.34

vcunat committed:
knot-dns: 3.2.7 -> 3.2.8

https://gitlab.nic.cz/knot/knot-dns/-/releases/v3.2.8

fabaff committed:
python311Packages.pyezviz: 0.2.1.3 -> 0.2.1.6

Diff: https://github.com/baqs/pyEzviz/compare/refs/tags/0.2.1.3...0.2.1.6

Changelog: https://github.com/BaQs/pyEzviz/releases/tag/0.2.1.6

r-ryantm committed:
python310Packages.ytmusicapi: 1.0.2 -> 1.1.0

vcunat committed:
gst_all_1.gst-plugins-rs: drop the raptorq test everywhere

I don't understand this, and it didn't seem anyone was willing
to look into the issue.
https://hydra.nixos.org/build/225312469/nixlog/1/tail

Commit:
separate minimal-bootstrap-sources from make-minimal-bootstrap-sources

Ericson2314 observed that although FODs break the build-time cycle,
they don't break the eval-time cycle.  This cycle must be broken in
order to compute the derivation hash.

make-minimal-bootstrap-sources does and always has depended on the
full nixpkgs (including fetchFromGitHub).  This commit completely
separates minimal-bootstrap-sources from it, so that
minimal-bootstrap-sources now does not depend on
make-minimal-bootstrap-sources (or nixpkgs) in any way.

minimal-bootstrap-sources is now a "bare Nix" derivation.  It is an
FOD whose builder always fails with an error message.

aaronjheng committed:
btcd: 0.23.3 -> 0.23.4

davidkna committed:
openai-whisper-cpp: enable CoreML support

Co-authored-by: wegank <9713184+wegank@users.noreply.github.com>

vcunat committed:
openmvg: fixup build

The build broke probably after cmake upgrade in b8ac5b6cfb,
but either way - this forcing of old C++ standard seemed bad.
https://hydra.nixos.org/build/225241222/nixlog/2/tail

Commit:
libnatpmp: fix install

Import an upstream patch (which actually predates 2023-04-23!) to
install natpmp_declspec.h, so that users of this library can be built.

vcunat committed:
basu: supply missing getent after PR #236458

https://hydra.nixos.org/build/225149294/nixlog/2/tail

Luflosi committed:
megapixels: 1.6.0 -> 1.6.1

https://gitlab.com/postmarketOS/megapixels/-/tags/1.6.1

r-ryantm committed:
python310Packages.plaid-python: 13.0.0 -> 14.1.0

r-ryantm committed:
oh-my-zsh: 2023-06-20 -> 2023-06-26

lilyinstarlight committed:
gst_all_1.gst-plugins-rs: add temporary fix for cargo-c test issue

vcunat committed:
Revert "mupdf: actually build and install the shared libraries version"

This reverts commit 3d4769a9e86a51117973281d8ab4328713c4c180.
This broke many packages; let's revert until that's resolved.
https://github.com/NixOS/nixpkgs/pull/237199#issuecomment-1607343333

r-ryantm committed:
python310Packages.pyglet: 2.0.7 -> 2.0.8

figsoda committed:
cargo-guppy: unstable-2023-06-19 -> unstable-2023-06-26

Diff: https://github.com/guppy-rs/guppy/compare/23b1c804cb4a6d9d38d093ca3400e90805d4c36c...c05b95997a127cebef76d3b238e0341086e59e32

figsoda committed:
cargo-hakari: 0.9.25 -> 0.9.26

Diff: https://github.com/guppy-rs/guppy/compare/cargo-hakari-0.9.25...cargo-hakari-0.9.26

Changelog: https://github.com/guppy-rs/guppy/blob/cargo-hakari-0.9.26/tools/cargo-hakari/CHANGELOG.md

figsoda committed:
cargo-nextest: 0.9.53 -> 0.9.54

Diff: https://github.com/nextest-rs/nextest/compare/cargo-nextest-0.9.53...cargo-nextest-0.9.54

Changelog: https://nexte.st/CHANGELOG.html

NickCao committed:
buildbot: unset SQLALCHEMY_SILENCE_UBER_WARNING

NickCao committed:
qjackctl: 0.9.10 -> 0.9.11

Diff: https://github.com/rncbc/qjackctl/compare/qjackctl_0_9_10...qjackctl_0_9_11

mweinelt committed:
firefox-beta-unwrapped: 115.0b7 -> 115.0b9

mweinelt committed:
firefox-devedition-unwrapped: 115.0b7 -> 115.0b9

figsoda committed:
fclones-gui: 0.1.2 -> 0.1.3

Diff: https://diff.rs/fclones-gui/0.1.2/0.1.3

swflint committed:
python310Packages.matplotx: init at 0.3.10 (#227075)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

wegank committed:
openai-whisper-cpp: refactor

NickCao committed:
saml2aws: 2.36.8 -> 2.36.9

Diff: https://github.com/Versent/saml2aws/compare/v2.36.8...v2.36.9

NickCao committed:
sssd: 2.9.0 -> 2.9.1

Diff: https://github.com/SSSD/sssd/compare/2.9.0...2.9.1

Changelog: https://sssd.io/release-notes/sssd-2.9.1.html

Tom-Hubrecht committed:
nixos/pixelfed: Fix missing php modules for pixelfed-horizon

Luflosi committed:
aaaaxy: 1.4.8 -> 1.4.18

https://github.com/divVerent/aaaaxy/releases/tag/v1.4.18

wegank committed:
proxysql: set platforms

figsoda committed:
typos: 1.15.5 -> 1.15.6

Diff: https://github.com/crate-ci/typos/compare/v1.15.5...v1.15.6

Changelog: https://github.com/crate-ci/typos/blob/v1.15.6/CHANGELOG.md

RaitoBezarius committed:
linux_6_4: init at 6.4

Email release: https://lwn.net/Articles/936310/
Kernel Newbies: https://kernelnewbies.org/Linux_6.4

linux_latest is bumped to linux_6_4 now.

NickCao committed:
postgresqlPackages.pgroonga: 3.0.7 -> 3.0.8, use msgpack-c

Diff: https://github.com/pgroonga/pgroonga/compare/3.0.7...3.0.8
Changelog: https://github.com/pgroonga/pgroonga/releases/tag/3.0.8

vcunat committed:
cockpit: specify --with-admin-group

Until PR #236458 it was doing auto-detection (!)
which chose "root" (at least sometimes):
https://hydra.nixos.org/log/wbxd753q70plgxjm3xdljky9a81692kk-cockpit-287.drv
Now it failed due to missing getent:
https://hydra.nixos.org/build/225207182/nixlog/2/tail

mweinelt committed:
python310Packages.pdm-backend: 2.1.0 -> 2.1.1

https://github.com/pdm-project/pdm-backend/releases/tag/2.1.1

emilylange commented on maintainers/maintainer-list.nix:
Please put this change (`maintainers/maintainer-list.nix`) into a separate commit titled `maintainers: add mccurdyc`.

So two commits in total:
1. `maintainers: add mccurdyc`
1. `gitrs: init at 0.3.6`

mccurdyc replied:
@emilylange thank you for the review and feedback!!

1. https://github.com/NixOS/nixpkgs/pull/238858/commits/2da53e573d8f615142f0ab6cbc159b5f99a5ba1b
2. https://github.com/NixOS/nixpkgs/pull/238858/commits/fba936bcfed146b9e3e6784a48709bda69ed1a01

emilylange commented on pkgs/tools/misc/gitrs/default.nix:
Formatting nit and the `meta.description` should not end with a period.
See https://nixos.org/manual/nixpkgs/unstable/#var-meta-description and https://github.com/NixOS/nixpkgs/blob/547cd96f10a3ac0126bc75d45b2337662df80f6a/CONTRIBUTING.md#submitting-changes

```patch
diff --git a/pkgs/tools/misc/gitrs/default.nix b/pkgs/tools/misc/gitrs/default.nix
index 9309e19dd26..4ce36855ef6 100644
--- a/pkgs/tools/misc/gitrs/default.nix
+++ b/pkgs/tools/misc/gitrs/default.nix
@@ -19,6 +19,12 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-+43XJroPNWmdUC6FDL84rZWrJm5fzuUXfpDkAMyVQQg=";
   };
 
+  cargoHash = "sha256-2TXm1JTs0Xkid91A5tdi6Kokm0K1NOPmlocwFXv48uw=";
+
+  nativeBuildInputs = [
+    pkg-config # for openssl
+  ];
+
   buildInputs = [ openssl.dev ]
     ++ lib.optionals stdenv.isDarwin [
     libiconv
@@ -26,17 +32,8 @@ rustPlatform.buildRustPackage rec {
     darwin.apple_sdk.frameworks.SystemConfiguration
   ];
 
-  # https://floxdev.com/blog/nixpkgs-contribution
-  # Nix recommends using SRI hashes, which are self-descriptive hashes, i.e. hashes that tell you which format they are in. You can use the nix hash to-sri $HASH --type sha256 to transform a sha256 hash to an SRI hash. You'll see that the result is prefixed with `sha256-`.
-
-  cargoHash = "sha256-2TXm1JTs0Xkid91A5tdi6Kokm0K1NOPmlocwFXv48uw=";
-
-  nativeBuildInputs = [
-    pkg-config # for openssl
-  ];
-
   meta = with lib; {
-    description = "A simple, opinionated, tool, written in Rust, for declaratively managing Git repos on your machine.";
+    description = "A simple, opinionated, tool, written in Rust, for declaratively managing Git repos on your machine";
     homepage = "https://github.com/mccurdyc/gitrs";
     license = licenses.mit;
     maintainers = with maintainers; [ mccurdyc ];
```

mccurdyc replied:
Fixed in https://github.com/NixOS/nixpkgs/pull/238858/commits/fba936bcfed146b9e3e6784a48709bda69ed1a01

Thank you for pointing this out!

emilylange replied:
Are you sure?
You might have lost that change when rebasing :)

mccurdyc replied:
Ahh sorry!! @emilylange should be fixed in https://github.com/NixOS/nixpkgs/pull/238858/commits/85c55158b4b26411574995ed9377b82ef2a50cad

Thank you for your patience!

emilylange replied:
Uhhm could you please remove that _floxdev_ comment as well and place `cargoHash` below `src` instead of `buildInputs`?

(See the patch I originally commented)

Thank you!

mweinelt committed:
python310Packages.async-upnp-client: 0.33.2 -> 0.34.0

https://github.com/StevenLooman/async_upnp_client/blob/0.34.0/CHANGES.rst

mweinelt committed:
python310Packages.ttp: 0.9.4 -> 0.9.5

https://github.com/dmulyalin/ttp/releases/tag/0.9.5

mweinelt committed:
zsh-powerlevel10k: 1.18.0 -> 1.19.0

https://github.com/romkatv/powerlevel10k/releases/tag/v1.19.0

wentasah committed:
fastddsgen: 2.5.0 -> 2.5.1

wegank committed:
ueberzugpp: 2.8.7 -> 2.8.8

mweinelt committed:
python310Packages.trainer: 0.0.26 -> 0.0.27

https://github.com/coqui-ai/Trainer/releases/tag/v0.0.27

mweinelt committed:
python310Packages.stdlib-list: 0.8.0 -> 0.9.0

https://github.com/pypi/stdlib-list/releases/tag/v0.9.0

mweinelt committed:
python310Packages.espeak-phonemizer: 1.3.0 -> 1.3.1

https://github.com/rhasspy/espeak-phonemizer/releases/tag/v1.3.1

SuperSandro2000 committed:
gopls: 0.12.2 -> 0.12.4 (#239950)

Diff: https://github.com/golang/tools/compare/gopls/v0.12.2...gopls%2Fv0.12.4

mweinelt committed:
python310Packages.django-timezone-field: 5.0 -> 5.1

Diff: https://github.com/mfogel/django-timezone-field/compare/5.0...5.1

malob committed:
signalbackup-tools: 20230625 -> 20230626

Diff: https://github.com/bepaald/signalbackup-tools/compare/20230625...20230626

wegank committed:
qt6.full: unlink qmake6

gaelreyrol committed:
scaphandre: init at 0.5.0 (#238462)

* scaphandre: init at 0.5.0

* Update pkgs/servers/scaphandre/default.nix

---------

Co-authored-by: Pol Dellaiera <pol.dellaiera@protonmail.com>

imincik committed:
grass: 8.2.1 -> 8.3.0 (#239901)

figsoda committed:
cargo-release: 0.24.10 -> 0.24.11

Diff: https://github.com/crate-ci/cargo-release/compare/v0.24.10...v0.24.11

Changelog: https://github.com/crate-ci/cargo-release/blob/v0.24.11/CHANGELOG.md

apfelkuchen6 committed:
texlive: fix licensing information for doc-only-packages (#239528)

* licenses: add opubl

* texlive: fix license for npp-for-context and lshort-german

This unbreaks evaluation for `texlive.combine { inherit (texlive) scheme-full;
pkgFilter = (_: true) }`. The problem doesn't manifest for the default pkgFilter
as `npp-for-context` is a doc-only-package (that by the way has recently been
removed from texlive).

Tillerino committed:
jetbrains: fix application icon (#239416)

My Desktop doesn't recognize the icon in ~/.local/share/pixmaps which is
how the icon would have to be found when doing a user profile install.

For me, it's hard to find information on this, but it seems like
share/pixmaps is something old and the share/icons location should be
used instead:

https://www.reddit.com/r/linuxquestions/comments/htg6c/comment/c1yfi0a/

https://github.com/spyder-ide/spyder/issues/6188

https://github.com/FreeSpacenav/spnavcfg/issues/9

In this PR we _add_ the icons location so that both are available for
compatibility.

marsam committed:
typescript: repackage using buildNpmPackage (#239189)

colonelpanic8 committed:
starship: Remove INSIDE_EMACS checks (#239039)

Many of the terminals supported inside emacs work perfectly fine with STARSHIP.
The TERM=dumb case already handles the tramp and eterm cases, so as far as I can
tell, this is basically just a check for the benefit of OLD versions of
term-mode (see
https://www.emacswiki.org/emacs/AnsiTerm#:~:text=Historically%2C%20'M%2Dx%20ansi%2Dterm,the%20older%20'C%2Dc'%20binding.,
which indicates that it also now handles colors)

SuperSandro2000 committed:
ArchiSteamFarm: 5.4.7.2 -> 5.4.7.3

drupol committed:
php83: init at 8.3.0alpha2 (#239151)

* php83: init at 8.3.0alpha2

* phpExtensions.blackfire: disable for php 8.3

* phpExtensions.apcu: patch for 8.3 only

keenanweaver committed:
ibm-plex: 6.2.0 -> 6.3.0

soywod committed:
comodoro: 0.0.8 -> 0.0.9

r-ryantm committed:
git-machete: 3.17.4 -> 3.17.6

xworld21 committed:
python3Packages.cson: init at 0.8

xworld21 committed:
python3Packages.speg: init at 0.3

xworld21 committed:
asymptote: make the python modules imported by xasy available at runtime

xworld21 committed:
asymptote: add Qt5 libraries and wrapper for xasy

xworld21 committed:
asymptote: enable lsp

mccurdyc committed:
maintainers: add mccurdyc

Signed-off-by: Colton J. McCurdy <mccurdyc22@gmail.com>

mccurdyc committed:
gitrs: init at 0.3.6

Signed-off-by: Colton J. McCurdy <mccurdyc22@gmail.com>
